components/SubscribeSection.tsx

```diff
@@ -18,26 +18,65 @@ import {
     Author,
     Tag
 } from '../lib/types/ghost-types';
+import { ChangeEvent, FormEvent, useState } from "react";
 
 export default function SubscribeSection() {
+
+    const [message, setMessage] = useState('');
+    const [email, setEmail] = useState('');
+
+    const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+        setEmail(event.target.value);
+        console.log(email);
+    }
+
+    const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+        const url = 'https://gobankless.ghost.io/members/api/send-magic-link'
+
+        const values = {
+            email,
+            emailType: `subscribe`,
+            labels: [],
+        };
+
+        try {
+
+            console.log('submitting email');
+            setMessage(`loading`);
+            await fetch(url, {
+                method: `POST`,
+                mode: `cors`,
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify(values),
+            });
+        } catch (e) {
+            setMessage(`error`);
+        }
+
+        setMessage(`success`)
+    };
+
     return (
         <chakra.div className="subscribe-section">
             <Flex className="subscribe-wrap">
                 <Heading as="h3">Subscribe to new posts.</Heading>
-                <Flex as="form" className="subscribe-form">
+                <chakra.form as="form" className="subscribe-form" onSubmit={ev => handleSubmit(ev)}>
                     <Input
                         type="email"
                         placeholder="Your email address"
                         aria-label="Your email address"
+                        value={email}
+                        onChange={handleChange}
                         isRequired
                     />
                     <Button className="global-button" type="submit">Subscribe</Button>
-                    <Box className="subscribe-alert">
-                        <chakra.small className="alert-loading"></chakra.small>
-                        <chakra.small className="alert-success"></chakra.small>
-                        <chakra.small className="alert-error"></chakra.small>
+                    <Box className={['subscribe-message', message ? message : ''].join(' ')}>
+                        <chakra.small className="alert-loading">Processing your application</chakra.small>
+                        <chakra.small className="alert-success">Great! Check your inbox and confirm your subscription</chakra.small>
+                        <chakra.small className="alert-error">There was an error sending the email.</chakra.small>
                     </Box>
```

Comment:
The three message types don't appear to be stylistically different to me, if that's the case I don't think rendering all three and using css to show the matching one is the most idiomatic approach. Instead you could define the message text values dictionary as a constant at top of the file and pull from that when rendering.
```jsx
// top of the file:
const ALERT_MESSAGES = {
  loading: "Processing your application",
  success: "Great! Check your inbox and confirm your subscription",
  error: "There was an error sending the email.",
}

<Box>
  <chakra.small>{ALERT_MESSAGES[message]}</chakra.small>
</Box>
```

I would also avoid using className or any globally defined styles where possible to take advantage of localized styles of the chakra architecture. Either add all the styles inline using the chakra style attributes or if you want to make something reusable then you should use the [Chakra factory](https://chakra-ui.com/docs/features/chakra-factory#attaching-styles) to define a component with attached styles:

```jsx
const AlertBox = chakra(Box, {
  baseStyle: {
      fontFamily: "four",
      fontSize: "12px",
      lineHeight: 1.1,
      position: "absolute",
      right: 0,
      bottom: "-38px",
      left: "20px",
      display: "none",
      width: "100%",
      margin: 0,
      padding: 0,
      color: "white"
  },
})
```

It's not even necessary to extract this component to its own file (you can just define it in the same component file that it's being used in if you like) but avoiding global styles make dead code elimination much simpler.